### ISSUE TITLE
Support date string in filter value for raw temporal field

### ISSUE THREAD

**Issue by kanitw** [vega/vega-lite#3958]
In https://github.com/vega/vega-lite/pull/3954, I'm making it work when timeUnit is specified.

**Comment by kanitw** [comment]
Previously in #1541, we discussed the possibility of adding `type` to filter (so we can infer the type when type = temporal but decided not to do it since filter doesn't really need to know all the type (except date) so type shouldn't be required, and when type isn't required, it won't be specified. 

1) One possible solution for this issue is to re-consider that option.

2) That said, another option that I prefer more is to introduce a second type of datetime objects for a date string. 

Basically, let's say `type DateTime = DateTimeDef | DateTimeStringDef` where
- `DateTimeDef` = original date time object with units as properties
- `type DateTimeStringDef = {datetime: string}` -- simply a wrapped of datetime string that can be parsed by Vega-Expression's `date()`. 

Any thoughts?
